node/source.sls

```diff
@@ -48,11 +54,42 @@ make-node:
     - cwd: /usr/src/node-v{{ version }}
     - name: make --jobs={{ make_jobs }}
 
+{% if grains['os'] in ['RedHat', 'CentOS'] %}
+make-install-node:
+  cmd.run:
+    - cwd: /usr/src/node-v{{ version }}
+    - name: make install DESTDIR=/usr/src/node-v{{ version }}/tmp
+    - onchanges:
+      - cmd: make-node
+
+fpm-node:
+  cmd.run:
+    - cwd: /usr/src/node-v{{ version }}
+    - name: fpm -s dir -t rpm -C /usr/src/node-v{{ version }}/tmp --name=node --version "{{ version }}" --iteration 1 --description "Node package v{{ version }} (fpm)" --force
+    - onchanges:
+      - cmd: make-install-node
+
+install-node:
+  pkg.installed:
+    - sources:
+      - node: /usr/src/node-v{{ version }}/node-{{ version }}-1.x86_64.rpm
+    - onchanges:
+      - cmd: fpm-node
+{% else %}
+
 install-node:
   cmd.run:
     - cwd: /usr/src/node-v{{ version }}
     - name: checkinstall --install=yes --pkgname=node --pkgversion "{{ version }}" --default
```

Comment:
@JulioECastro this needs to be pulled out for centos, `checkinstall` is not installed